cpp/src/cuda/cuda_matrix_vector_multiply.cu

```diff
@@ -20,17 +20,18 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 #include "include/cuda_matrix_vector_multiply.h"
-
+#define BLOCK_SIZE 7
```

Comment:
A BLOCK_SIZE of 7 is not a good idea. It should always be multiple of 32 (warp size)